dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/sam/SamPermissionsImpl.java

```diff
@@ -321,11 +321,11 @@ private void addPolicy(ResourcesApi resourcesApi, String resourceId, String poli
 
     @Override
     public boolean canDoAction(User user, Workflow workflow, Role.Action action) {
-        ResourcesApi resourcesApi = getResourcesApi(user);
-        String encodedPath = encodedWorkflowResource(workflow, resourcesApi.getApiClient());
         try {
+            ResourcesApi resourcesApi = getResourcesApi(user);
+            String encodedPath = encodedWorkflowResource(workflow, resourcesApi.getApiClient());
             return resourcesApi.resourceAction(SamConstants.RESOURCE_TYPE, encodedPath, SamConstants.toSamAction(action));
-        } catch (ApiException e) {
+        } catch (ApiException | CustomWebApplicationException e) {
```

Comment:
seems like we should log the exception ... though as INFO since this may be legit behaviour if the user doesn't have permission to view. I worry a little this may suppress an error message that may legitimately have been useful to return to the user. 

I wonder if a less impactful fix may be to only catch exceptions with that specific google error message and let the rest through.

Reply:
(also not literal. A more classy way to do it instead of string matching is to maybe throw a custom subclass of `CustomWebApplicationException` and catch that)